src/onboarding/onboarding-integrations.ts

```diff
@@ -35,20 +35,20 @@ class OnboardingIntegrations extends SubscribeMixin(LitElement) {
 
   @state() private _entries: ConfigEntry[] = [];
 
-  @state() private _discoveredDomains = new Set<string>();
+  @state() private _discoveredDomains?: Set<string> = undefined;
 
   public hassSubscribe(): (UnsubscribeFunc | Promise<UnsubscribeFunc>)[] {
     return [
       subscribeConfigFlowInProgress(this.hass, (messages) => {
-        messages.forEach((message) => {
-          if (
-            message.type === "removed" ||
-            HIDDEN_DOMAINS.has(message.flow.handler)
-          ) {
-            return;
-          }
-          this._discoveredDomains.add(message.flow.handler);
-        });
+        this._discoveredDomains = new Set(
```

Comment:
This will remove any previous `_discoveredDomains` and looking at the [core PR](https://github.com/home-assistant/core/pull/142459/files) it seems wrong. Core sends a list first, then keeps sending "added"/"removed" events